etc/openqa/openqa.ini

```diff
@@ -136,3 +136,9 @@ blacklist = job_grab job_done
 # Specify the number of screenshot ID ranges (with a size as configured by screenshot_cleanup_batch_size) to process in a single Minion
 # job (reduce to avoid Minion jobs from running very long and possibly being interrupted, increase to reduce the number of Minion jobs)
 #screenshot_cleanup_batches_per_minion_job = 450
+
+[job_settings_ui]
+# Specify the directories which contains test suite settings
+# Directories should be under CASEDIR root path or under the
+# /data folder of the CASEDIR
+#keys_to_render_as_links=schedule,autoyast_sle15
```

Comment:
I am sorry if we come to a misunderstanding here. You are starting work on a new feature that sounds good in theory but has a lot of implications. The specific problem becoming apparent here is that etc/openqa/openqa.ini is the config file for an *openQA instance* that can run multiple test distributions independantly. Think for example about https://openqa.opensuse.org/ running tests from the distri "opensuse" from https://github.com/os-autoinst/os-autoinst-distri-opensuse/ as well as "obs" from https://github.com/os-autoinst/os-autoinst-distri-obs and "openqa" from https://github.com/os-autoinst/os-autoinst-distri-openQA/ . IMHO does not make sense that the instance admin would need to configure something what is essentially dictated by the test distribution. Now I do not have the ultimate answer as again: You are opening a new field, which we appreciate but we just have to be open that we still change the architecture.

How about the following: openQA already renders automatic URLs for everything that "looks like" a URL, e.g. http://foo.bar is rendered into an `<a href…`. For the settings that you want to cover the challenge is that we only have relative paths which need the internal information if for example you use the top folder "$casedir/data/" or "$casedir/schedule" which so far is only a convention within os-autoinst-distri-opensuse. Right now I see three options:

1. We specify one folder as "special" and every variable that references that folder in the name together with the path separation token "/" we prepare as link to the according file within $casedir. Example `AUTOYAST=data/autoyast_opensuse/opensuse_minimal.xml` would be rendered with a link to `$casedir/data/autoyast_opensuse/opensuse_minimal.xml`. Same for schedule files if the folder "schedule" would be moved
2. Use special setting names, similar to what probably Martchus referred to with *keys* but using convention rather than configuration. Example anything with suffix `_REL_URL` would be relative to $datadir so that e.g. `AUTOYAST_REL_URL=data/autoyast_opensuse/opensuse_minimal.xml` would link accordingly to `$casedir/data/autoyast_opensuse/opensuse_minimal.xml`.
3. Turn all relative URLs already into absolute URLs either statically within os-autoinst-distri-opensuse or within openQA at schedule time (but then again we would need to identify which settings to transform)

For now I prefer 1. WDYT?

Reply:
it is hard to say. All of them sounds good and reasonable. 
1. This requires to move folders into `data/`. From my POV this would be more convenience but i am not sure if `schedule` and `test_data` belong there. Also this will require some changes in `os-autoinst-distri-opensuse` for the scheduler as well as for the relative paths of the variables (AUTOYAST for example does not contain data/. it looks like `AUTOYAST=autoyast_opensuse/opensuse_minimal.xml`)

2. This makes more clear what @Martchus also suggested but still i can not see clear how the implementation will look like. Either with convention or configuration. i assume that ` _REL_URL` will be used somehow to the template. But how can it be used in such a way that makes it to know that this should be render as a link is unknown to me.

3. i am not fun of this solution as i can see restrictions and other implications.

Reply:
Not sure whether it makes sense to make one folder special. Option 2. sounds better. Not sure how option 3. would solve the problem.

Option 4.: We could allow rendering `file://` URLs similarly to how we render `https://` URLs already. This means you would have to use URLs and we need to ensure that the places where the path is used can cope with the URL format. We could also use a more specific custom scheme for the URLs, e.g. `test-file://`.

But I'm also not sure what the problem with my approach to configure the keys is. Of course an openQA can be used to test multiple test distributions. The admin will need to provide checkouts of these distributions under `/var/lib/openqa/tests`, right? So I see no problem in also making that configuration for these distributions if wanted.

Reply:
@mkittler the admin does not need to check out distributions when you think about git checkouts at least. The setting in /etc is just too generic and too hidden. As a compromise the product settings or "Medium Settings" would be a better fit because they are linked to individual distributions. But of course already within osd we have a lot of duplication within medium settings. And the only other point where we communicate settings from the test distribution to openQA is vars.json which we don't want to parse on the settings page.

Reply:
The product or medium settings are used to generate job settings. They would not be a fitting place for controlling openQA's UI behavior despite their scope might match.